server-common/src/main/java/org/apache/kafka/server/common/TestFeatureVersion.java

```diff
@@ -23,8 +23,8 @@ public enum TestFeatureVersion implements FeatureVersion {
 
     // TEST_1 released right before MV 3.7-IVO was released, and it has no dependencies
     TEST_1(1, MetadataVersion.IBP_3_7_IV0, Collections.emptyMap()),
-    // TEST_2 released right before MV 3.8-IVO was released, and it depends on this metadata version
-    TEST_2(2, MetadataVersion.IBP_3_8_IV0, Collections.singletonMap(MetadataVersion.FEATURE_NAME, MetadataVersion.IBP_3_8_IV0.featureLevel()));
+    // TEST_2 released right before MV 4.0-IV0 was released, and it depends on this metadata version
+    TEST_2(2, MetadataVersion.IBP_4_0_IV0, Collections.singletonMap(MetadataVersion.FEATURE_NAME, MetadataVersion.IBP_4_0_IV0.featureLevel()));
```

Comment:
@jolshan I need help understanding this test setup. As far as I could understand, this `TEST_2` feature was meant to be released in a future release, given we 3.8 is the current one, I moved it to 4.0. But I think this is not a robust test. Probably I'm missing something else.

Reply:
I plan to update this in a future PR. Sorry it is not easy to understand now. If you can push it to 4.0, that would be good for now.